src/OrchardCore/OrchardCore.Liquid.Abstractions/LiquidPropertyAccessor.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using System.Threading.Tasks;
+using Fluid.Values;
+
+namespace OrchardCore.Liquid
+{
+    /// <summary>
+    /// Can be used to provide a factory to return a value based on a property name 
+    /// that is unkown at registration time. 
+    /// 
+    /// e.g. {{ LiquidPropertyAccessor.MyPropertyName }} (MyPropertyName will be passed as the identifier argument to the factory)
+    /// </summary>
+    public class LiquidPropertyAccessor
```

Comment:
Is it still necessary?

Reply:
You mean because we now have the ability to use factories?

I initially wondered the same, I don't *think* we can get substitute this class for the factory values. This class allows us to effectively look 2 properties down the chain (i.e. `ContentItemId.SomeId`, whereas the factory would only allow us to look at the next property on (`ContentItemId`).

We might be able to simplify this class a little however, I'll investigate.

Reply:
Yep, I think we still need this class. 

We are already using the new factory functionality to return an instance of this class (https://github.com/OrchardCMS/OrchardCore/pull/1548/files/b7969fcee5ff5fc004c435f57cc342df72b65675#diff-a868b4d5cdf08f24e42293fab511e5afR26). This class effectively allows us to chain factories. You could even chain instances of this class together to created nested factories.